### PR TITLE
moved custom analytics to kick in after React is done

### PR DESCRIPTION
**Pull request by Pomax**
Closes https://github.com/mozilla/foundation.mozilla.org/issues/4583

### CODE REVIEW

File: source/js/buyers-guide/analytics-events.js [hunk 1]
```diff
@@ -8,52 +8,16 @@ function getQuerySelectorEvents(pageTitle, productName) {
...
       action: `donate tap`,
       label: `${pageTitle} donate header`
     },
-    "#donate-button-footer": {
+    ".donate-banner a.btn.btn-secondary": {
       category: `buyersguide`,
       action: `donate tap`,
       label: `${pageTitle} donate footer`
     },
-    "#nav-social-button-fb": {
```
Comment (Pomax) [inline source/js/buyers-guide/analytics-events.js:21]: we removed this functionality a while back and never removed these query selectors

File: source/js/buyers-guide/analytics-events.js [hunk 2]
```diff
@@ -77,43 +41,20 @@ function getQuerySelectorEvents(pageTitle, productName) {
       action: `opinion submitted`,
       label: `opinion on ${productName}`
     },
-    "#product-social-button-fb": {
```
Comment (Pomax) [inline source/js/buyers-guide/analytics-events.js:80]: same here, we removed them in favour of the single copy-and-share link

File: source/js/buyers-guide/analytics-events.js [hunk 3]
```diff
@@ -77,43 +41,20 @@ function getQuerySelectorEvents(pageTitle, productName) {
...
-      transport: `beacon`
-    },
-    "#privacy-policy-link": {
+    "a.privacy-policy-link": {
       category: `product`,
       action: `privacy policy link tap`,
       label: `policy link for ${productName}`,
       transport: `beacon`
     },
-    "#reading-level-link": {
```
Comment (Pomax) [inline source/js/buyers-guide/analytics-events.js:104]: this is not a thing anymore.

File: source/js/buyers-guide/analytics-events.js [hunk 4]
```diff
@@ -146,6 +87,8 @@ const ProductGA = {
         let eventData = querySelectorEvents[querySelector];
 
         elements.forEach(e => setupElementGA(e, eventData));
+      } else if (!querySelectorEvents[querySelector].optional) {
+        console.error(`cannot find ${querySelector}`);
```
Comment (cadecairos) [inline source/js/buyers-guide/analytics-events.js:91]: nice.

File: source/js/buyers-guide/analytics-events.js [hunk 5]
```diff
@@ -77,43 +41,20 @@ function getQuerySelectorEvents(pageTitle, productName) {
...
-    },
 
     // product updates
     ".product-update-link": {
       category: `product`,
       action: `update article link tap`,
       label: `update article link for ${productName}`,
-      transport: `beacon`
+      transport: `beacon`,
+      optional: true
```
Comment (mmmavis) [inline source/js/buyers-guide/analytics-events.js:57]: `optional` doesn't seem to be a [GA Event option](https://developers.google.com/analytics/devguides/collection/analyticsjs/events). Can we add inline comment to explain what it is for?

File: source/js/buyers-guide/bg-main.js [hunk 6]
```diff
@@ -85,6 +84,9 @@ let main = {
       // Record that we're done, when we're really done.
       Promise.all(apps).then(() => {
         window[`bg-main-js:react:finished`] = true;
+
+        // bind custom analytics only once everything's up and loaded
+        AnalyticsEvents.init();
```
Comment (mmmavis) [inline source/js/buyers-guide/bg-main.js:89]: 👍

**Pull request merged**